DR-2099: implement upsert, delete endpoints for catalog metadata

pshapiro4broad opened the pull request:
Also changed upsert payload so entire body is metadata

pshapiro4broad committed:
DR-2099: implement upsert, delete endpoints for catalog metadata

Also changed upsert payload so entire body is metadata

fboulnois commented on src/test/java/bio/terra/app/controller/SearchApiControllerTest.java:
```diff
@@ -0,0 +1,67 @@
+    verify(snapshotMetadataDao).deleteMetadata(id);
+  }
+
+  private void verifyAuthApi(UUID id) {
```
How is this expected to work with a random uuid?

pshapiro4broad replied:
It verifies that the `iamService` API was called with the uuid provided. So the uuid is random for each test run but is the same between the call to the endpoint (via `mvc.perform()`) and the iam service being verified.

pshapiro4broad replied:
To answer the question about how does the code not fail because I'm not mocking `when(snapshotMetadataDao).putMetadata()` etc using the random uuid, it's not necessary because the mocked DAO APIs are `void`, so the calling code isn't dependent on the results. Mockito records all calls to the mocks, so the test can use `verify()` to find out if the mock was called with the arguments test expects.

pshapiro4broad committed:
- update test data to better reflect actual data
- change SnapshotSearchMetadataDaoTest to Unit, as it only uses a DB connection

fboulnois reviewed (approved):
lgtm, one question:

pshapiro4broad committed:
Verify result of `PUT` endpoint
